generator/src/main/java/com/scottlogic/deg/generator/inputs/MainConstraintReader.java

```diff
@@ -27,8 +27,12 @@ public Constraint apply(
             throw new InvalidProfileException("Constraint is null");
         }
 
-        if (dto.is != null) {
-            ConstraintReader subReader = this.atomicConstraintReaderLookup.getByTypeCode(dto.is);
+        if (dto.is == null) {
+            throw new InvalidProfileException("Couldn't recognise null from is constraint");
+        }
+
+        if (dto.is != dto.undefined) {
```

Comment:
```suggestion
        if (dto.is != ConstraintDTO.undefined) {
```

Access the static field via the class not the instance